contessa/runner.py

```diff
@@ -44,29 +45,47 @@ def run(
         self.ensure_table(quality_check_class)
 
         rules = self.build_rules(normalized_rules)
-        objs = self.do_quality_check(quality_check_class, rules, context)
+        objs = self.do_quality_checks(quality_check_class, rules, context)
 
         self.insert(objs)
 
+    def get_context(self, check_table: Table, context: Optional[Dict] = None) -> Dict:
+        """
+        Construct context to pass to executors.
+        note: could be overridden in Executor.get_context() method.
+        """
+        ctx_defaults = {
+            "table_fullname": check_table.fullname,
+            "task_ts": datetime.now(),  # todo - is now() ok ?
+        }
+        new_context = context or {}
+        for k, v in ctx_defaults.items():
```

Comment:
Just out of curiosity: Does this mean that we do not want "unknown" keys in the context?

Reply:
I take the passed context by client and then just add default values in case user doesn't passed the key-value. means that client is responsible for context we just provide "some defaults"

Reply:
Oh, I missed the line before 🤦‍♂ Otherwise why not use simple ‘update’ on dict?